internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -658,7 +673,7 @@ func flattenKustoClusterSku(sku *clusters.AzureSku) []interface{} {
 }
 
 func flattenKustoClusterVNET(vnet *clusters.VirtualNetworkConfiguration) []interface{} {
-	if vnet == nil {
+	if vnet == nil || *vnet.State == clusters.VnetStateDisabled {
```

Comment:
I don't think the `state` check is necessary anymore?

```suggestion
	if vnet == nil {
```